src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/IModelMetadataProvider.cs

```diff
@@ -12,4 +13,12 @@ public interface IModelMetadataProvider
 
         IEnumerable<ModelMetadata> GetMetadataForProperties(Type modelType);
     }
+
+    // TODO: Could we take a breaking change and add this to the IModelMetadataProvider
+    // interface? If not, it means that anyone who's built a custom IModelMetadataProvider
+    // will not get the new parameter validation feature.
+    public interface IExtendedModelMetadataProvider : IModelMetadataProvider
+    {
+        ModelMetadata GetMetadataForParameter(ActionDescriptor actionDescriptor, ParameterDescriptor parameter);
```

Comment:
Would it be possible to use `ParameterInfo` here? The rest of the MM APIs only deal with reflection not with MVC's action types.

Reply:
Yes, done.